src/ImageSharp/Formats/Jpeg/Components/Decoder/HuffmanTable.cs

```diff
@@ -62,20 +55,7 @@ internal unsafe struct HuffmanTable
         /// <param name="values">The huffman values</param>
         public HuffmanTable(ReadOnlySpan<byte> codeLengths, ReadOnlySpan<byte> values)
         {
-            this.isConfigured = false;
-            Unsafe.CopyBlockUnaligned(ref this.Sizes[0], ref MemoryMarshal.GetReference(codeLengths), (uint)codeLengths.Length);
```

Comment:
As I recall this copy exists because the input span is allocated once and reused for each Huffman table.

Reply:
Which one? `codeLengths`? It may be unique for each table. Right now ctor copies input spans and then builds actual table representation during `Configure()` call - that's the only reason it copies first span param to internal array. While calling this method a couple of times with short-circuit `return` is not bad, it's impossible to pass workspace spans from allocator this way. That's why I want to build table in the ctor but that remote executor crash...

I hope it's okay if I can use github actions for some testing. That random crash is very strange and I'm not sure I can debug it locally.

Reply:
Ah yes, The reason for the lazy invocation was to avoid building the tables to allow the validation of successive scan parameters for progressive images. See `HuffmanScanDecoder.CheckProgressiveData`. This is the same thing that libjpeg and libjpeg-turbo does. https://github.com/libjpeg-turbo/libjpeg-turbo/blob/main/jdhuff.c#L77

Reply:
Also yes, experiment away. Maybe disable the codecov run first though to save time.

Reply:
Hm, that check seems valid but table building is not that expensive, especially compared to exception path imo. Should I still remove `Configure` and move everything to the ctor for simplicity and reduced `stackalloc`s?

Reply:
Let’s give it a go yeah!